.github/workflows/shared-wildfly-build-and-test.yml

```diff
@@ -146,34 +146,46 @@ jobs:
         run: JAVA_HOME=$JAVA_HOME_${{ matrix.java }}_${{ runner.arch }} mvn -U -B -ntp clean install ${{ inputs.build-arguments }} ${{ inputs.test-arguments }}
         shell: bash
         # We need this due to long file names on Windows: https://github.com/actions/upload-artifact/issues/240
-      - name: Find reports
-        if: failure()
+      - name: Find reports on Windows
+        if: ${{ failure() && contains(matrix.os, 'windows') }}
         run: |
           {
             echo 'SUREFIRE_REPORTS<<EOF'
             find . -path '**/surefire-reports/*.xml'
             echo EOF
           } >> "$GITHUB_ENV"
         shell: bash
-      - name: Find server logs
-        if: failure()
+      - name: Find server logs on Windows
+        if: ${{ failure() && contains(matrix.os, 'windows') }}
         run: |
           {
             echo 'SERVER_LOGS<<EOF'
             find . -path '**/server.log'
             echo EOF
           } >> "$GITHUB_ENV"
         shell: bash
-      - name: Upload Test Reports on Failure
+      - name: Upload Test Reports on Failure on Windows
         uses: actions/upload-artifact@v4
-        if: failure()
+        if: ${{ failure() && contains(matrix.os, 'windows') }}
         with:
           name: surefire-reports-${{ matrix.os }}-${{ matrix.java }}
           path: ${{ env.SUREFIRE_REPORTS }}
-      - name: Upload Server Logs on Failure
+      - name: Upload Server Logs on Failure on Windows
         uses: actions/upload-artifact@v4
-        if: failure()
+        if: ${{ failure() && contains(matrix.os, 'windows') }}
         with:
           name: server-logs-${{ matrix.os }}-${{ matrix.java }}
           path: ${{ env.SERVER_LOGS }}
+      - name: Upload Test Reports on Failure
+        uses: actions/upload-artifact@v3
```

Comment:
@jfdenise Does this and the one below need to be v3?

Reply:
Typo. Thanks @bstansberry , fixed.